File: portfolio-server/src/main/java/com/maksudsharif/portfolioserver/model/SkillGroup.java
package com.maksudsharif.portfolioserver.model;

import java.io.Serializable;
import java.util.List;
import lombok.Builder;
import lombok.Data;

@Data
@Builder
public class SkillGroup implements Serializable
{

  private String name;
  private List<Skill> skills;
  private SkillTypeView type;
}

File: portfolio-client/src/app/components/model/skillGroup.ts
import {Skill} from './skill';
import {SkillItemType} from './skillItemType';

export class SkillGroup {
  name: string;
  skills: Skill [];
  type: SkillItemType;


  constructor(name: string, skills: Skill[], type: SkillItemType) {
    this.name = name;
    this.skills = skills;
    this.type = type;
  }

  static from(name: string, skills: any[], type: string) {
    const finalSkills = [];
    for (const skill of skills) {
      finalSkills.push(Skill.of(skill.name, skill.level, skill.years));
    }
    return new SkillGroup(name, finalSkills, SkillItemType[type]);
  }
}
